tests/Games/NexusMods.Games.RedEngine.Tests/RedModDeployToolTests.cs

```diff
@@ -0,0 +1,109 @@
+using System.Reactive.Linq;
+using DynamicData;
+using DynamicData.Binding;
+using FluentAssertions;
+using Microsoft.Extensions.DependencyInjection;
+using NexusMods.Abstractions.Loadouts;
+using NexusMods.Games.RedEngine.Cyberpunk2077;
+using NexusMods.Games.RedEngine.Cyberpunk2077.SortOrder;
+using NexusMods.Games.TestFramework;
+using NexusMods.Paths;
+using R3;
+using ReactiveUI;
+using StrawberryShake.Extensions;
+using Xunit.Abstractions;
+
+namespace NexusMods.Games.RedEngine.Tests;
+
+public class RedModDeployToolTests : ACyberpunkIsolatedGameTest<Cyberpunk2077Game>
+{
+    private readonly RedModDeployTool _tool;
+
+    public RedModDeployToolTests(ITestOutputHelper helper) : base(helper)
+    {
+        _tool = ServiceProvider.GetServices<ITool>().OfType<RedModDeployTool>().Single();
+    }
+    
+    [Fact]
+    public async Task LoadorderFileIsWrittenCorrectly()
```

Comment:
There's a small caveat with these tests.

The tests here are also asserting the determinism of the underlying mod installer, i.e.
`RedModInstaller.ExecuteAsync`, since one of the archives, `several_red_mods.7z` contains
multiple mods.

The installer *does* sort the metadata files, in `infosList.OrderBy(x => x.InfoJson.Name)` @ `RedModInstaller.cs`, which is good 👍.
However, a note in the source code still needs to be made somewhere that changing that constitutes a possible breaking change.

Either in this method (preferably here since if test fails someone would look here) or in the relevant installer code.
Just want to avoid someone doing unnecessary debugging if possible.

Reply:
Good point, I went over the code and I think we should be good on that front due to this line here:
https://github.com/Nexus-Mods/NexusMods.App/blob/7046eb9583516b4573dd64ad248e12d46ff91bc0/src/Games/NexusMods.Games.RedEngine/Cyberpunk2077/SortOrder/RedMod/RedModSortableItemProvider.cs#L72

I sort the RedModGroups by name before I iterate over them during the Synchronization operation, so new entries should get added in consistent order, regardless of installer order.

I'm pretty sure the installer does a single commit, which means that we get all the new red mods in a single batch, which then gets sorted. 

So I guess this can fail if the installer is changed to do multiple commits?

Reply:
Yup, what we have works, just need to note it down somewhere in a comment.